ssr/webpack.config.js

```diff
@@ -24,67 +50,113 @@ const config = {
+                ],
+              },
+            ],
           },
           {
-            loader: "file-loader",
-            options: {
-              emitFile: false,
-              publicPath: "/",
-              name: "static/media/[name].[hash].[ext]",
-            },
+            rules: [
+              {
+                test: /\.tsx?$/,
+                exclude: /node_modules/,
+                use: [
+                  {
+                    loader: "ts-loader",
+                    options: {
+                      transpileOnly: true,
+                    },
+                  },
+                ],
+              },
+              {
+                test: /\.svg$/,
+                use: [
+                  {
+                    loader: "@svgr/webpack",
+                    options: {
+                      prettier: false,
+                      svgo: false,
+                      svgoConfig: {
+                        plugins: [{ removeViewBox: false }],
+                      },
+                      titleProp: true,
+                      ref: true,
+                    },
+                  },
+                  {
+                    loader: "file-loader",
+                    options: {
+                      emitFile: false,
+                      publicPath: "/",
+                      name: "static/media/[name].[hash].[ext]",
+                    },
+                  },
+                ],
+                issuer: {
+                  and: [/\.(ts|tsx|js|jsx|md|mdx)$/],
+                },
+              },
+              {
+                test: [/\.avif$/, /\.bmp$/, /\.gif$/, /\.jpe?g$/, /\.png$/],
+                type: "asset",
+                parser: {
+                  dataUrlCondition: {
+                    maxSize: 0,
+                  },
+                },
+                generator: {
+                  emit: false,
+                  publicPath: "/",
+                  filename: "static/media/[name].[hash][ext]",
+                },
+              },
+              { test: /\.(css|scss)$/, loader: "ignore-loader" },
+            ],
           },
         ],
-        issuer: {
-          and: [/\.(ts|tsx|js|jsx|md|mdx)$/],
-        },
-      },
-      {
-        test: [/\.avif$/, /\.bmp$/, /\.gif$/, /\.jpe?g$/, /\.png$/],
-        type: "asset",
-        parser: {
-          dataUrlCondition: {
-            maxSize: 0,
-          },
-        },
-        generator: {
-          emit: false,
-          publicPath: "/",
-          filename: "static/media/[name].[hash][ext]",
-        },
       },
-      { test: /\.(css|scss)$/, loader: "ignore-loader" },
     ],
   },
-  externals: nodeExternals(),
+  // // TODO/BUG: this was never working, when we did `cd ssr && webpack`
+  // // but now we do `webpack --config ssr/webpack.config.js` the yari
+  // // node_modules dir gets picked up, and all of that stuff is `require()`ed
+  // // which then makes `dist/main.js` no longer a ESM-compatible module.
```

Comment:
Some of column A some of column B: what broke things was running `webpack --config ssr/webpack.config.js` rather than `cd ssr && webpack` - note the directory node is running from there is different, and ssr doesn't have a "node_modules" dir, so `nodeExternals()` was essentially a no-op before, whereas now it's trying to keep all of node_modules external, but with `require()`s rather than `import`s.

It's possible to make it ESM compatible, and we'll want to do that for local dev because it speeds things up so much, but as a future incremental fix.